### PR TITLE
fix(inputs.procstat): Collect swap via /proc/$pid/smaps

### PR DESCRIPTION
**Pull request by powersj**
Currently `proc.MemoryInfo()` claims to collect rss, vms, as well as swap, data, stack, and locked information. This functions reads from /proc/$PID/statm. However the gopsutil function only returns two values, rss and vsm, the first and second value from that file. Additionally, the man page for proc says that instead of statm users should look to smaps or smaps_rollup:

```
If accurate values are required, use /proc/pid/smaps or /proc/pid/smaps_rollup instead
```

This change will keep rss and vms from the /proc/$PID/statm file and introduce a call to /proc/$PID/smaps

* size: total size
* pss: process's proportional share of this mapping
* shared_clean/shared_dirty: the number of clean and dirty shared pages in the mapping
* private_clean/private_dirty: the number of clean and dirty private pages in the mapping
* referenced: the amount of memory currently marked as referenced or accessed
* anonymous: the amount of memory that does not belong to any file
* swap: how much would-be-anonymous memory is also used, but out on swap

As the previous values were always zero, users should not have any expectations on the data, stack, or locked fields. This does mean users can now get the swap field as well as the new fields.

fixes: #8557

### CODE REVIEW

**Review by srebhan — COMMENTED** [review]
One simplification to get rid of the additional `memmap_notlinux.go` file...

File: plugins/inputs/procstat/procstat.go [hunk 1]
```diff
@@ -225,16 +225,15 @@ func (p *Procstat) addMetric(proc Process, acc telegraf.Accumulator, t time.Time
...
 		fields[prefix+"memory_rss"] = mem.RSS
 		fields[prefix+"memory_vms"] = mem.VMS
-		fields[prefix+"memory_swap"] = mem.Swap
-		fields[prefix+"memory_data"] = mem.Data
-		fields[prefix+"memory_stack"] = mem.Stack
-		fields[prefix+"memory_locked"] = mem.Locked
 	}
 
+	collectMemmap(proc, prefix, fields)
+
```
[the remark this answers is not in the document]
Comment (powersj) [reply plugins/inputs/procstat/procstat.go:236]: In gopsutil, Darwin does not have `MemoryMapsWithContext`. Probably worth an upstream issue and fix...

[the remark this answers is not in the document]
Comment (srebhan) [reply plugins/inputs/procstat/procstat.go:236]: Hmm isn't this covered by the [BSD part of the implementation](https://github.com/shirou/gopsutil/blob/89575bdf5ad71dbaa0970f426c8f90e3bd759aa4/process/process_bsd.go#L59)?

[the remark this answers is not in the document]
Comment (powersj) [reply plugins/inputs/procstat/procstat.go:236]: ah good catch, I only looked for it in the darwin source files

[the remark this answers is not in the document]
Comment (powersj) [reply plugins/inputs/procstat/procstat.go]: I've pushed an update, let's see what CI says

[the remark this answers is not in the document]
Comment (powersj) [reply plugins/inputs/procstat/procstat.go]: Now I remember why I did this: https://app.circleci.com/pipelines/github/influxdata/telegraf/17805/workflows/b57957ca-4e54-4461-802b-c3e2831974a4/jobs/275344

**Review by srebhan — APPROVED** [review]
Nice. Thank you @powersj!

**Pull request merged**